audit_log/middleware.py

```
@@ -29,6 +34,158 @@ def _enable_audit_log_managers(instance):
+    """
+    Wraps a synchronous signal handler to run in a thread pool for ASGI contexts.
+    
+    Django signals are synchronous, so they will call handlers synchronously even in
+    ASGI contexts. This wrapper ensures the handler runs in a thread pool to avoid
+    blocking the event loop.
+    """
+    def wrapper(*args, **kwargs):
+        # Import here to avoid issues if not in async context
+        import asyncio
+        try:
+            loop = asyncio.get_running_loop()
```

Ruff (0.13.1)

Local variable `loop` is assigned to but never used

Remove assignment to unused variable `loop`

(F841)